### DISCUSSION TITLE
Primary key error when using sa_column=mapped_column

### DISCUSSION THREAD

**Discussion by simkessy** [fastapi/sqlmodel#970] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from __future__ import annotations

from datetime import date, datetime
from typing import List, Optional

from sqlalchemy import BigInteger, Column, Date, DateTime, ForeignKeyConstraint, Index, Integer, LargeBinary, String, Table, Text, text
from sqlalchemy.orm import mapped_column
from sqlalchemy.dialects.mysql import ENUM, INTEGER, LONGTEXT, MEDIUMBLOB, TEXT, TINYINT, TINYTEXT
from sqlmodel import Field, Relationship, SQLModel


class ContactDetails(SQLModel, table=True):
    __tablename__ = 'contact_details'

    id: Optional[int] = Field(
        default=None, sa_column=mapped_column("id", Integer, primary_key=True)
    )
    user_id: int = Field(sa_column=mapped_column('user_id', Integer))
    pinner_id: Optional[int] = Field(default=None, sa_column=mapped_column('pinner_id', BigInteger))
    address: Optional[str] = Field(default=None, sa_column=mapped_column('address', TINYTEXT))
    city: Optional[str] = Field(default=None, sa_column=mapped_column('city', TINYTEXT))
```

### Description

I created a model and when I try to import it I get an error about the primary key the examples I'm seeing look like this online

```
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/util/deprecations.py", line 281, in warned
    return fn(*args, **kwargs)  # type: ignore[no-any-return]
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/orm/mapper.py", line 853, in __init__
    self._configure_pks()
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/orm/mapper.py", line 1639, in _configure_pks
    raise sa_exc.ArgumentError(
```

### Operating System

Linux

### SQLModel Version

0.0.19

### Python Version

3.8.16

**Comment by spazm** [comment]
SQLmodel has special handling for sa_column that requires it to be a Column that has not been extended to support MappedColumn.  I think this is a bug / oversight from the transition to sqlalchemy 2.0

PR #896 fixes this but was closed by its author deleting the branch.  I opened a new PR #1143 from the branch and added tests.  Your example works with the sqlmodel version in the PR.

Community:  How can I raise the visibility of my PR and see if it fits with the project?
